File: apps/server/aetherblog-service/blog-service/src/main/java/com/aetherblog/blog/entity/Tag.java
package com.aetherblog.blog.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * 标签实体
 * 
 * @ref §6.1 - 核心表结构 (V2 增强版)
 */
@Data
@Entity
@Table(name = "tags", indexes = {
    @Index(name = "idx_tags_slug", columnList = "slug"),
    @Index(name = "idx_tags_post_count", columnList = "post_count")
})
public class Tag {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(nullable = false, unique = true, length = 50)
    private String slug;

    /**
     * 标签描述 (V2新增)
     */
    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 20)
    private String color = "#6366f1";

    /**
     * 文章数量 (缓存字段)
     */
    @Column(name = "post_count", nullable = false)
    private Integer postCount = 0;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: apps/server-go/internal/model/tag.go
package model

import "time"

// Tag 对应数据库 `tags` 表，表示用于文章分类标注的标签。
// 标签与文章之间为多对多关系，支持颜色标识以便在 UI 中快速区分。
type Tag struct {
	ID          int64     `db:"id"`
	Name        string    `db:"name"`        // 标签显示名称
	Slug        string    `db:"slug"`        // URL 友好的唯一标识符
	Description *string   `db:"description"` // 标签描述，可为空
	Color       string    `db:"color"`       // 标签颜色（十六进制），用于 UI 视觉区分
	PostCount   int       `db:"post_count"`  // 使用该标签的文章数量（缓存值）
	CreatedAt   time.Time `db:"created_at"`  // 记录创建时间
	UpdatedAt   time.Time `db:"updated_at"`  // 记录最后更新时间
}
